source/slang/slang-check-decl.cpp

```diff
@@ -9308,8 +9308,161 @@ namespace Slang
             this, funcDecl, attr, DeclAssociationKind::PrimalSubstituteFunc);
     }
 
+    static int32_t findLevelsOfInheritance(AggTypeDecl* decl, int32_t inheritanceLevel, List<AggTypeDecl*>& linearInheritanceHierarchy)
+    {
+        // We need to assume multiple `InheritanceDecl` are possible per `decl` member list.
+        int32_t inheritanceDepth = inheritanceLevel;
+        for (auto maybeInheritance : decl->members)
+        {
+            auto inheritance = as<InheritanceDecl>(maybeInheritance);
+            if (!inheritance)
+                continue;
+            auto isValidInheritance = as<DeclRefType>(inheritance->base.type);
+            if (!isValidInheritance || !isValidInheritance->getDeclRef())
+                continue;
+            auto isInherited = as<AggTypeDecl>(isValidInheritance->getDeclRef().getDecl());
+            if (!isInherited)
+                continue;
+            linearInheritanceHierarchy.add(isInherited);
+            int32_t depthFound = findLevelsOfInheritance(isInherited, inheritanceLevel-1, linearInheritanceHierarchy);
```

Comment:
I'm curious why you use negative value for the level?
Can't the logic become?
```
 int32_t depthFound = findLevelsOfInheritance(isInherited, inheritanceLevel + 1, linearInheritanceHierarchy);
inheritanceDepth = max(depthFound, inheritanceDepth)
```

Reply:
I thought negative made sense (since we use this logic for getting a constructor overload ranking), but in hindsight positive is better and easier to read, I will change the code accordingly.